src/main/kotlin/nl/info/zac/productaanvraag/ProductaanvraagEmailService.kt

```diff
@@ -0,0 +1,89 @@
+import nl.info.client.zgw.shared.ZGWApiService
+import nl.info.client.zgw.zrc.model.generated.MedewerkerIdentificatie
+import nl.info.client.zgw.zrc.model.generated.NatuurlijkPersoonIdentificatie
+import nl.info.client.zgw.zrc.model.generated.NietNatuurlijkPersoonIdentificatie
+import nl.info.client.zgw.zrc.model.generated.OrganisatorischeEenheidIdentificatie
+import nl.info.client.zgw.zrc.model.generated.Zaak
+import nl.info.zac.mail.MailService
+import nl.info.zac.mail.model.MailAdres
+import nl.info.zac.mail.model.getBronnenFromZaak
+import nl.info.zac.productaanvraag.exception.EmailAddressNotFoundException
+import nl.info.zac.productaanvraag.exception.InitiatorNotFoundException
+import nl.info.zac.productaanvraag.exception.MailTemplateNotFoundException
+import nl.info.zac.util.AllOpen
+import nl.info.zac.util.NoArgConstructor
+import nl.info.zac.zaak.ZaakService
+
+@ApplicationScoped
+@NoArgConstructor
+@AllOpen
+class ProductaanvraagEmailService @Inject constructor(
+    private val zgwApiService: ZGWApiService,
+    private val klantClientService: KlantClientService,
+    private val zaakService: ZaakService,
+    private val mailService: MailService,
+    private val mailTemplateService: MailTemplateService,
+) {
+    fun sendEmailForZaakFromProductaanvraag(
+        zaakFromProductaanvraag: Zaak,
+        zaakafhandelParameters: ZaakafhandelParameters
+    ) {
+        zaakafhandelParameters.automaticEmailConfirmation?.takeIf { it.enabled }?.let { automaticEmailConfirmation ->
+            val to = extractInitiatorEmail(zaakFromProductaanvraag)
+            val mailTemplate = mailTemplateService
+                .findMailtemplateByName(automaticEmailConfirmation.templateName)
+                .orElseThrow {
+                    MailTemplateNotFoundException(
+                        "No mail template found with name: '${automaticEmailConfirmation.templateName}'"
+                    )
+                }
+            val mailGegevens = MailGegevens(
+                MailAdres(automaticEmailConfirmation.emailSender, null),
+                MailAdres(to, null),
+                MailAdres(automaticEmailConfirmation.emailReply, null),
+                mailTemplate.onderwerp,
+                mailTemplate.body,
+                null,
+                true
+            )
+            mailService.sendMail(mailGegevens, zaakFromProductaanvraag.getBronnenFromZaak())
+            zaakService.setOntvangstbevestigingVerstuurdIfNotHeropend(zaakFromProductaanvraag)
+        }
+    }
+
+    private fun extractInitiatorEmail(createdZaak: Zaak): String {
+        val identificatie = zgwApiService.findInitiatorRoleForZaak(createdZaak)?.betrokkeneIdentificatie
+            ?: throw InitiatorNotFoundException(
+                "No initiator rol or identification found for the zaak: '${createdZaak.uuid}'"
+            )
+        return when (identificatie) {
+            is MedewerkerIdentificatie -> extractEmail(identificatie.identificatie)
```

Comment:
A medewerker can never be an initiator. It does not make sense to check for that here.